common/build/eslint-plugin-fluid/src/test/mockFiles/no-restricted-tags-imports/fileWithExceptionImports.ts

```diff
@@ -3,4 +3,4 @@
  * Licensed under the MIT License.
  */
 
-import { exceptionInternalFunction, alphaFunction2 } from "./exceptionFile.ts";
```

Comment:
Not sure why this is automatically chaning from `ts` to `js` just by moving the directory.

Reply:
In general imports shouldn't have ts extensions. js/cjs/mjs are ok. I imagine you probably told VSCode to auto-update imports when the file moved, and if so it did the right thing here.

Reply:
https://github.com/microsoft/FluidFramework/blob/22b05b09c0881a04633fae4712c794da77909411/common/build/eslint-config-fluid/src/test/mockFiles/no-restricted-tags-imports/fileWithExceptionImports.ts#L6

I think using `.ts` extension is acceptable as this was the case from the `eslint-config-fluid` package?

Reply:
Not really. It doesn't fail because we're not actually running those files, just linting them, but Typescript is complaining about the import even now in my local:

![image](https://github.com/microsoft/FluidFramework/assets/716334/9b132a1a-ac70-4744-a9c1-74777d754189)

Best to just use `.js` even if these files are never really executed.

Reply:
Left comment below about changing it to `.ts`.

Reply:
I still don't understand why we have `.ts` here. We don't use this setting in our general build settings and I don't think we intend to.
I am requesting `allowImportingTsExtensions` to be removed from tsconfig. If it should stay, please explain why.

Reply:
Just importing with a `.js` extension causes the test to fail because there's no such `.js` file. Note this project doesn't do a Typescript build. The actual tests are written in `.js` files (`src/test/enforce.../*.js`), and all these `.ts` files inside `src/test/mockFiles/` are just collateral for testing the custom linting rules, and they're never compiled. It feels right to me that the rules are tested "on top of" `.ts` files since that's what they would normally operate on in the wild.

For the purpose of the test, we just need that one of the `.ts` files is importing from another one that exports things with different API-levels, and I guess writing both in TS was the more natural thing to do. If `exceptionFile` was a `.(c/m)js` file directly and we imported from it with that extension, things probably would be fine too? We'd just need to remove the typing from it. Not sure I prefer that so much over the current state of things, maybe expanding on the comment in line 6 about why we do it.

I imagine a setup exists where we compile the collateral files during the build, so then when the tests for the custom lint rules run they have both the source and the built versions, but I'm not sure what exactly we gain from such a setup.

Reply:
There is a test tsconfig.json - not part of the normal build, but it will be.
Of course there isn't a .js file, but .js is the way import specs should be written ... without `allowImportingTsExtensions`. Typescript and related evaluators should understand that a .js path may resolve to .ts or .d.ts in a Typescript context. There is no requirement that the .js file exists during Typescript linting. (Our build used to require tsc run before eslint, but that was wrong - I think I removed that requirement. If not, I will get it removed ;) )
Note that we do not use `allowImportingTsExtensions` in our normal tsconfig settings. The settings here should be similar to the minimal requirements for our build. (So, no `allowImportingTsExtensions`.)
Technically we could lint a .js file for the tags associated with the imports by looking at the corresponding .d.ts, but I think that is uncommon use case and we can ignore it.
I am concerned that the related rules are not really doing their job in the real world.

Reply:
So the issue here might be that the ["test" tsconfig.json](https://github.com/microsoft/FluidFramework/blob/main/common/build/eslint-plugin-fluid/src/test/tsconfig.json) should look more like one of our normal tsconfig.json's, and that might let the dummy TS project which gets linted in this package's tests to `import from ... .js"` like we do everywhere else?

Of note, I hesitate to just call it "test tsconfig.json" because that makes me think of "the tsconfig json used for the build of the tests", which at least today, this one is not; neither source nor tests have a build step in this package right now. The tsconfig.json is used for a dummy TS project "created programmatically" [for the ESLint instance we create and exercise during tests](https://github.com/microsoft/FluidFramework/blob/22d9380428c1d79067b65c2843d9e3efb5b5a27a/common/build/eslint-plugin-fluid/src/test/enforce-no-member-release-tags/enforce-no-member-release-tags.test.js#L11-L27).

> I am concerned that the related rules are not really doing their job in the real world.

`no-member-release-tags` does trigger today:

![image](https://github.com/user-attachments/assets/61665af2-937e-4830-8c52-e4014548afce)

`no-restricted-tags-imports`... seems not to be used today, ha.

Reply:
Yeah - it should be the "examples" tsconfig file. (I didn't go as far as to suggest relocating the file in #22005, but we definitely should when making this package use typescript.)

Reply:
I am not saying `no-member-release-tags` won't trigger today. I am questioning whether it is triggering everywhere we want it to. I have not looked at the rule to try to understand it.